### ISSUE TITLE
Confidence Estimates for Semantic Segmentation Tasks

### ISSUE THREAD

**Issue by biancazadrozny** [IBM/terratorch#566]
Automatically estimating the confidence levels of semantic segmentation predictions is important, as it can enhance accuracy and allow final users to have better insights into whether or not they can trust model results.

TerraTorch users should be able to add an (optional) confidence estimation step to semantic segmentation tasks to get confidence maps during inference (in addition to the standard outputs of the model).

One such method has been implemented in https://github.com/ESA-PhiLab/CAS_Confidence_Assessment, so I suggest we should integrate it into TerraTorch.

**Comment by Joao-L-S-Almeida** [comment]
According to the [original paper](https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=10759697), that is a post-processing operation (page 2) that will be performed just after the segmentation head, right ?
In this way we could optionally output the CAS map to separated files in the same we way do for logits, for example.

**Comment by romeokienzler** [comment]
@biancazadrozny still relevant? then we reopen...otherwise closing due to inactivity
